common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -409,6 +466,131 @@ bool mon_wlan_hal_dwpal::sta_link_measurements_11k_request(const std::string &st
     return true;
 }
 
+bool mon_wlan_hal_dwpal::dcs_scan_trigger(int dwell_time_msec,
+                                          const std::vector<unsigned int> &channel_pool)
+{
+    LOG(DEBUG) << __func__ << " received on interface=" << m_radio_info.iface_name;
+
+    //build scan parameters
+    ScanParams dcs_scan_params;
+    memset((void *)&dcs_scan_params, 0, sizeof(dcs_scan_params)); //must to ensure freq are zero
+    size_t out_len                                = 0;
+    unsigned char out_data[NL_MAX_REPLY_BUFFSIZE] = {'\0'};
+    sScanCfgParams fg, new_fg;   //foreground scan param
+    sScanCfgParamsBG bg, new_bg; //background scan param
```

Comment:
Instead of two copies, we can just use the one copy and separately remember the two values we override (nitpick).

Reply:
I believe that having two copies of sScanCfgParams & sScanCfgParamsBG, makes the code easier to read and understand, and besides that, if we want to change other parameters later on having the two copies would make it easier to do so